### ISSUE TITLE
Check of Modifications from Sept 2021 and open features

### ISSUE THREAD

**Issue by RalfWerner** [termux/termux-float#37]
### check scenarios
The *Taskbar** App [examination](https://github.com/termux/termux-x11/pull/19#issuecomment-908235805), was very sobering if I compare this with [PiP](https://github.com/RalfWerner/integrated-process/issues/2#issuecomment-733700709) on my [SM-P610](https://user-images.githubusercontent.com/45426380/125047883-b241a300-e09f-11eb-9952-09bd1f22f905.mp4), in which nothing is missing and that works very stable. That's why I've never used **T:Float**. For comparison and to fix possibly bugs of **TfT** (**T**askbar-**f**loating-**T**ermux), I've this App (self-build with last updates from @agnostic-apollo) now installed on my Gigaset and tries to understand this App.

**Float** Readme is not very informative. The permission query after the installation is good (would be good also at **T:Widget**).
First, I describe the check scenarios. The **Float** App is positioned on the Start Screen between Termux and **Widget**, so **Termux** can be started with all three Apps.
The _normal_ [OTS](https://github.com/termux/termux-float/issues/37#issuecomment-916777123) open **X11** with some XClients so that XServer and `sshd` Services exist and uniquely the process environment is logged. When **Float** starts from the start screen again, a fixed quarity picture appears without Extra-Keys (EK) but with OS-KB. The process list is supplemented by a bash that does _not_ appear like widget processes in the Drawer process list.
![grafik](https://user-images.githubusercontent.com/45426380/131866441-f069053f-be8c-4b26-96ad-12348dde1229.png)
- Shot1. Unfortunately, it is not a _nomal_ session like you get with **Taskbar** and have different _long tap_ behavior: In addition to missing EK, there is **no** `DRAWER` possibility and _long tap_ (inside - not at the top) allows move and scallation (Pinch) with appropriate instructions (good) but **no clipboard**. The movement beyond the screen, as with taskbar does not work.
- Shot2 I then extended the picture portrait, started from Widget _check_ (**X11**) and the turned phone and have still no chance to remove OS-KB. You see: no prompt visible, because the end of **Float**-Picture is outside of the screen. Move of **Float** is only up to the half screen possible but **Float** could be minimized to a circle (first symbol in border) 

*) Taskbar on the one hand and Floating/Picture or Android Windows on the other hand are _different topics._ In the Taskbar App and with Android-with-PiP contained therein (11/SM-P610), both comes only together. On various App/Task Functions (Recent-Task-Key) gestures and much more, which has every phone, _I do not go on details_ here and also the topic session/widget-lists currently has _no relation_ to **Float**. Here are two more shots, from my Gigaset (above) without Split/PiP capability and the SM-P610:
![grafik](https://user-images.githubusercontent.com/45426380/132223499-ea66a01b-05b7-4e23-b2cd-ea12b6848fef.png)
- Shot1 (Gigaset) is with Tap on **T:Float**, then start of `t c` in **Float**, OS-KB turn off/on in **X11** (with Extra-Keys=EK) and change to **Hackers KB**. Then swich _Key-log on_ in `check` and tap on `SHIFT+ALT+CTRL` (blue). Here **Float**-Picture is covered by OS-KB (log is hidden but _Back-Key_ remove it) even after _pinch_ to the smalest high of Picture and more critical with _non floating_ **gboad-KB**_. **X11**-EK is _unusable in **Float**_! 
- Shot2 (SM-P610) Since SM-Android already covers all the features of **Float** and **Taskbar** and this can be used for almost all apps, you do not need both apps. Here **Float** is only for comparison and since the OS already has several task bars, I have not installed this App.
**Termux** is here startet from _site-Taskbar_ (moved to PiP) and [OTS](https://github.com/termux/termux-float/issues/37#issuecomment-916777123) open **X11** here in Background. Previously the File-App** was opened in Picture and is now shrinked to the _blue Icon ([also a Taskbar by tap](https://github.com/RalfWerner/integrated-process/issues/2#issuecomment-733700709)) between 1/2 over the _non-floating_ OS-KB.  **Float** deprives themself to the Picture/Task Management of the OS, so it is initially unclear, what happened, if you **tap on 5** of OS-KB - what are you **supposed?** Both EK are _not_ usable for **Float** but OS-KB for all three Apps.
### conclusion and possible features
It remains: **Float** is 100% covered by SM-OS, a reduced variant of Taskbar-Termux-Floating and without parallel **Termux**, barely useful. Here is **Float** optically hardly to distinguish from termux. Should as announced all future Android versions have behaved the SM-OS, is unnecessary the **Float** development. Potentials:

- The missing/double EK administration should be replaced by a common, as with OS-KB incusive **Float** and **X11**.
- **Float** could also be managed through **Widget**, which would also solve the problems with Drawer/Session list.
### More checks, special effects and misconduct with **TfT** and **Float**:
**Float** only have the opportunity to move the picture inside the screen at the Picture-size, but it can scaled greater than this with pinch. This has the effect that the picture can _not_ be moved or only in _one_ direction. Then, on the right edge, the contents are cut off the content instead of wraps and in height is the prompt often under the real screen and almost always under the non-floating OS-KB. The rotation of the phones often leads to similar effects. As in old Termux versions, the penetrient _popup_ of the OS-KB can not be prevented.
**Float** does not take into account the reduction of the residual surface by EK+OS-KB. However, mostly through skilful inner and outer pinch zooms, the relavant part of Termux can be brought into the visible area. The effects can most often be bypassed _long Tap pinch/move_. Who that seems to be complotted, should _try_ **TfT** to return back to **Float**.
In **TfT** a complete **Termux** with _nomal_ behavior is used but as soon as the **X11** screen is touched or with _Back-Key_, Termux will removed fron screen and has to reopen by Taskbar*. Move of **TfT** is only one time after _Back-Key_ and _reopen: possible.

**Comment by RalfWerner** [comment]
**OTS**=One-Touch-Start  is the Termux behavior that starts with _one tap_ on the App-Icon (startscreen) all (closed) processes for **X11** and other via `.bashrc`. Unfortunately, the description was [deleted](https://github.com/WMCB-Tech/termux-x11/discussions/4#discussion-3557430), to this and many other topics. So here again with:
``` sh
echo '[ -z "$1" ]&& . uu.sh . .; t=$PREFIX/tmp X=Xwayland XDG_RUNTIME_DIR=$t WM=fluxbox
c="am start -n com.termux" z="$c/.app.TermuxActivity" y="$c.x11/.MainActivity"; typeset -i i=0
[ -n "$(pgrep m.termux.window)" ]&& z="$c/.window.TermuxFloatActivity";alias t="bash .bashrc"
export DISPLAY=:0; while [ -z "$(pgrep -o $X)" -a $i -lt 2 ];do $y;sleep 1;$X :0>$t/TX 2>&1&
  $z; i+=1;done; [ "$1" = 0 ]&& $y; [ "$1" = 2 ]&& $z;[ -z "$(pgrep -o $WM)" ]&& $WM&'>>.bashrc
```
will OTS set up or supplemented. Since **Float** is a separate app, it is displayed in parallel/over **Termux** but only once (no Window Manager like `fluxbox` in **X11** above). OTS uses as X11-Toggle `$y/$z` by `t 0|2`, but `$z` would be replaced by **Float** (line 3), if _tap_ on this App and would be displayed _over_ **X11**. Otherwise **Termux** start standalone or by **TfT**, also be used simultaneously _over_ **X11** but very [weak](). Without _Split_ and _floating_ **Termux** X11-Toggle-Key should be part of EK.

The script `uu.sh` for OTS not necessary, but does some useful things like device check and _double-u_ (`ln -s ../usr u;alias u="bash uu.sh"`). So with `u ky` the EK line(s) is to produced/toggled. Here the script extract with X11-Key (`t 0`) In line 5:
``` sh
ky)t=~/.termux;[ ! -e $t ]&& mkdir $t; t+=/termux.properties; d=display; k=extra-keys;p=popup
 m=macro;sk="hide-soft-keyboard-on-startup=true soft-keyboard-toggle-behaviour=enable/disable"
 lf="{key:LEFT,$p:BKSP}"; ed="$p:{$m:'CTRL d',$d:d}"; eh="{key:END,$p:HOME}"; o=`grep one $t`
 nx="{key:ENTER,$p:{$m:'CTRL n',$d:'next'}}"; tp="{key:TAB,$p:PASTE}"; ek=$k-style=arrows-all
 kd="{key:KEYBOARD,$p:DRAWER}";x="{$d:X11,$m:'t SPACE 0 ENTER',$p:{macro:'ob ENTER',display:OB}}"
 mc="{$m:'mc SPACE i ENTER',$d:mc,$p:F10}" cd="{$m:CTRL,$d:'^',$ed}"; m="#"
 if [ "$o" != "" ];then k+="=[[ESC,SHFT,$kd,{key:'/',$p:{$d:'\\\\\\\\',macro:BACKSLASH}},{key:\
  '-',$p:'|'},UP,$nx,HOME,PGUP,$x],[ALT,$cd,$tp,{key:BKSP, $p:DEL},$lf,DOWN,RIGHT,END,PGDN,$mc]]"
 else k+="=[[$cd,$tp,$lf,{key:RIGHT,$p:{macro:'o ENTER',display:off}},{key:UP,$p:PGUP},\
  {key:DOWN,$p:PGDN},$eh,$nx,$kd,$mc,$x]]"; m="#one"
 fi ;printf '%s\n' $m $ek shortcut.next-session=ctrl+n $sk enforce-char-based-input=true "$k">$t
 termux-reload-settings; exit ;;
```
To toggle back you could open `aterm` in `fluxbox` and use command `t 2` or other actions (e.g `check`). The current state of **X11**-EK contains bugs that do not allow proper update and implementation.

**Comment by agnostic-apollo** [comment]
Resize is working on Android 11 avd as well.

https://user-images.githubusercontent.com/31106828/148833499-8fb49bbf-6642-427c-a486-74748cebf063.mp4

>The ip function is no longer included in the bootstrap 

It was just a symlink to `/system/bin/ip`, but has been removed with [`2812170`](https://github.com/termux/termux-packages/commit/2812170). Directly call path.

>The path nl=~/../../lib exist not anymore! I use it to analyze *.apk

Like I have told you before somewhere, termux does not create that path.

Closing issue since who knows what this is about. Open a new issue and fill template for any problems.

**Comment by RalfWerner** [comment]
> Resize is working on Android 11 avd as well (**for you**).

Thanks for the reaction. One of the **Float** Bugs you have already been reproduced in your own video(0:08) . As far as _resize_ is concerned, I expected your behavior on my AVD as on all other devices. _**Is your log with my (above) comperable?**_
I've created another video (over Starscreen on Pixel2 with Android 11) in which you can see my behavior right next to yours.

https://user-images.githubusercontent.com/45426380/148918903-c0b1725a-d758-49ab-b603-1a84e9d319f1.mp4

Your _resize_ Start at 0:00/0:01 (0:07) and my first attempt at 0:04 and the second at 0:07 (with shift positioning of the eccentric center). Previously (0:02), I had made a _move_ without Ctrl and both only _move -**Why?**_

Either you have already made an _Resize_ before the video, another AVD used or otherwise specified the _"Init size"_ (larger:"Report issues.." without clipping/line break). As can be seen in my shots above, I can only use **one size** as here on AVD devices.
If you use the `ip` function, as in my video (with line break at input and output) missing in your edition without exit circle (0:08 and larger font) probably the rest of the line and is no longer reproduced.
If you further enlarge the Zoom Vertical, the **Termux** prompt is suddenly no longer available. Since I can not reproduce it on AVD, I did it on my Gigaset - Shots above.

I now assume that _**you know these two bugs**_ and I do not have to describe them again!
Another important feature is the _missing_ **EK** (12 times in this issue above) in **Float**. If you are planning the integration of **Float** in **Termux**, which would require [Fullscreen](https://github.com/termux/termux-app/pull/1923) (also [here](https://github.com/termux/termux-x11/issues/29#issuecomment-916850563)) there are most of my concerns resolved. 
> Like I have told you before somewhere, termux does not create that path.

I know! but here the first time on AVD-Devices (it is my 11th device). On _old devices_, I could create this _symlink_ by my self with `pm` but this funktion is missing on _new devices_ or do you _**know a workarrond?**_

**Comment by agnostic-apollo** [comment]
Drag outer two dots left-right/up-down further from the center dot, outside of screen. If you only do it near center dot, window will just move instead of resize.

`termux-float` should preferably be integrated into `termux-app` if possible and I see not much point in working more on it. I don't know when I'll have time for integration or just adding extra keys, lot of other higher priority things to do.

**Comment by RalfWerner** [comment]
> Drag outer two dots left-right/up-down further from the center dot, outside of screen. If you only do it near center dot, window will just move instead of resize.
![grafik](https://user-images.githubusercontent.com/45426380/148941762-d0203a66-3c53-4626-9847-b004afd51a03.png)

_**So?**_ Unfortunately, AVD _**still only moves**_ in this case only the _entire window_. I've spent many hours in front of my PC, tracted my mouse and bent my fingers. Also in this case, the _**moves**_ lead to the following status: 
![grafik](https://user-images.githubusercontent.com/45426380/148942928-44afea7b-29db-435c-8491-645b16dd9b88.png)
Again the log of _this_ Device (for compare with yours):
```
D:\Android>cd sdk
emulator: Android emulator version 30.8.4.0 (build_id 7600983) (CL:N/A)
handleCpuAcceleration: feature check for hvf
added library vulkan-1.dll
Failed to open /qemu.conf, err: 2
HAX is working and emulator runs in fast virt mode.
emulator: INFO: GrpcServices.cpp:315: Started GRPC server at 127.0.0.1:8554, security: Local
emulator: INFO: EmulatorAdvertisement.cpp:93: Advertising in: C:\Users\th80wr\AppData\Local\Temp\avd\running\pid_18616.ini
emulator: emulator window was out of view and was recentered
emulator: INFO: QtLogger.cpp:68: Critical: Failed to load https://maps.googleapis.com/maps/api/mapsjs/gen_204?csp_test=true: The 'Access-Control-Allow-Origin' header has a value 'qrc://' that is not equal to the supplied origin. Origin 'qrc://' is therefore not allowed access. (qrc:/html/js/common.js:0, (null))
```
> `termux-float` should preferably be integrated into `termux-app` if possible and I see not much point in working more on it. I don't know when I'll have time for integration or just adding extra keys, lot of other higher priority things to do.

:) Very good Idea! What is your scheduling? Except for my Gigaset (no _SPLIT_) I use it nowhere really and to the much better **PiP** (popup view) function of Samsung, there are elaborate errors. I've edit EK-Remarks [above](https://github.com/termux/termux-float/issues/37#issuecomment-1009250552)
